src/fusets/mogrp.py

```diff
@@ -0,0 +1,138 @@
+    for x, y in itertools.product(range(x_size), range(y_size)):
+
+        X_vec = []
+        Y_vec = []
+        Y_mean_vec = []
+        Y_std_vec = []
+
+        for ind in range(noutput_timeseries):
+            X_tmp = time_in[ind]
+            Y_tmp = data_in[ind][:, x, y]
+            X_tmp = X_tmp[~np.isnan(Y_tmp), np.newaxis]
+            Y_tmp = Y_tmp[~np.isnan(Y_tmp), np.newaxis]
+            X_vec.append(X_tmp)
+            Y_vec.append(Y_tmp)
+            del X_tmp, Y_tmp
+
+        if np.size(Y_vec[master_ind]) > 0:
+
+            # Data Normalization
+            for ind in range(noutput_timeseries):
+                Y_mean_vec.append(np.mean(Y_vec[ind]))
+                Y_std_vec.append(np.std(Y_vec[ind]))
+                Y_vec[ind] = (Y_vec[ind] - Y_mean_vec[ind]) / Y_std_vec[ind]
+
+            # Multi-output train and test sets
+            Xtrain = X_vec
+            Ytrain = Y_vec
+
+            nsamples, npixels = Xtest.shape
+            noutputs = len(Ytrain)
+
+            for i_test in range(nt):
+                Yp = np.zeros((nsamples, noutputs))
+                Vp = np.zeros((nsamples, noutputs))
+                # K            =  GPy.kern.RBF(1, lengthscale=32.9169)
+                K = GPy.kern.Matern32(1, lengthscale=32.9169)  # Use RBF or Matern32 as kernels
+                LCM = GPy.util.multioutput.LCM(input_dim=1,
+                                               num_outputs=noutputs,
+                                               kernels_list=[K] * noutputs, W_rank=1)
+                model = GPy.models.GPCoregionalizedRegression(Xtrain, Ytrain, kernel=LCM.copy())
+                model['.*Mat32.var'].constrain_fixed(1.)
+                # model['.*rbf.var'].constrain_fixed(1.)
+                if not np.isnan(Ytrain[1]).all():
+
+                    try:
+                        # if trained_model is None:
+                        model.optimize()
+                        list_tmp = [model.param_array]
+
+                        for _ in range(noutput_timeseries):
+                            list_tmp.append(eval('model.sum.ICM' + str(_) + '.B.B'))
+                        out_model[x][y] = list_tmp
+
+
+                    except:
+                        out_qflag[x, y] = False
+                        continue
+
+                    for out in range(noutputs):
+                        newX = Xtest.copy()
+
+                        newX = np.hstack([newX, out * np.ones((newX.shape[0], 1))])
+                        noise_dict = {'output_index': newX[:, -1:].astype(int)}
+                        Yp[:, None, out], Vp[:, None, out] = model.predict(newX, Y_metadata=noise_dict)
+
+                    if i_test == 0:
+
+                        for ind in range(noutput_timeseries):
+                            out_mean[ind][:, None, x, y] = (Yp[:, None, 0] * Y_std_vec[ind] + Y_mean_vec[ind]) / Nt
```

Comment:
@msalinero  (FYI: I copied this from AI4FOOD_OpenEO_MOGPR_S1S2.ipynb )
this line (and 3 other lines) use a variable `Nt`  that is not defined anywhere.

Reply:
as discussed with @msalinero L `Nt` should be `nt` here